### DISCUSSION TITLE
Is there a way to share a mutation result?

### DISCUSSION THREAD

**Discussion by atsixian** [TanStack/query#3058] [Q&A] [answered]
Mutaions have no key and it's also stated in [TkDodo's blog](https://tkdodo.eu/blog/mastering-mutations-in-react-query#:~:text=Another%20difference%20is%20that%20mutations%20don%27t%20share%20state%20like%20useQuery%20does.%20You%20can%20invoke%20the%20same%20useQuery%20call%20multiple%20times%20in%20different%20components%20and%20will%20get%20the%20same%2C%20cached%20result%20returned%20to%20you%20%2D%20but%20this%20won%27t%20work%20for%20mutations.) that it's not shared:
> Another difference is that mutations don't share state like useQuery does. You can invoke the same useQuery call multiple times in different components and will get the same, cached result returned to you - but this won't work for mutations.

Is there any encouraged pattern to get a mutation's result(success/failure) from elsewhere? I know we can leverage React and put the mutation into context or pass it as props. But I'm curious why you made the design decision not to have something like `useMutation(mutationKey)` to share the result globally.

Thank you!

**Comment by TkDodo** [comment, marked as answer]
mutations work differently from queries, because every mutation becomes a different observer once you execute `mutate`. So even if we make the result sharable, it would likely only be the "latest" result. There is an open issue however: 
- https://github.com/tannerlinsley/react-query/issues/2304

and I do have some ideas on rewriting mutations completely to maybe make that happen :)

if you only need the loading state of a mutation - `useIsMutating` can help

**Reply by atsixian** [reply]
Thanks for the quick response! Looking forward to the progress.
